deploy/config/crds/submarineraddon.open-cluster-management.io_submarinerconfigs.yaml

```diff
@@ -41,107 +47,209 @@ spec:
+                description: CredentialsSecret is a reference to the secret with a
+                  certain cloud platform credentials, the supported platform includes
+                  AWS, GCP, Azure, ROKS and OSD. The submariner-addon will use these
+                  credentials to prepare Submariner cluster environment. If the submariner
+                  cluster environment requires submariner-addon preparation, this
+                  field should be specified.
                 properties:
                   name:
-                    description: 'Name of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names TODO: Add other useful fields. apiVersion, kind, uid?'
+                    description: 'Name of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+                      TODO: Add other useful fields. apiVersion, kind, uid?'
                     type: string
                 type: object
               gatewayConfig:
-                description: GatewayConfig represents the gateways configuration of the Submariner.
+                description: GatewayConfig represents the gateways configuration of
+                  the Submariner.
                 properties:
                   aws:
-                    description: AWS represents the configuration for Amazon Web Services. If the platform of managed cluster is not Amazon Web Services, this field will be ignored.
+                    description: AWS represents the configuration for Amazon Web Services.
+                      If the platform of managed cluster is not Amazon Web Services,
+                      this field will be ignored.
                     properties:
                       instanceType:
                         default: m5n.large
-                        description: InstanceType represents the Amazon Web Services EC2 instance type of the gateway node that will be created on the managed cluster. The default value is `m5n.large`.
+                        description: InstanceType represents the Amazon Web Services
+                          EC2 instance type of the gateway node that will be created
+                          on the managed cluster. The default value is `m5n.large`.
                         type: string
                     type: object
+                  azure:
+                    description: Azure represents the configuration for Azure Cloud
+                      Platform. If the platform of managed cluster is not  Azure Cloud
+                      Platform, this field will be ignored.
+                    properties:
+                      instanceType:
+                        default: m5n.large
```

Comment:
It looks like `m5n.large` is not a valid instanceType for Azure. Can you please check.

Reply:
Dome